src/udp_collector.cpp

```diff
@@ -261,16 +260,24 @@ bool UDPCollector::handle_one()
         return true;
     }
 
-    log_hex_printf(logio, Level::Debug, rxbuf, nrx, "UDP Rx %d, %s -> %s @%u (%s)\n",
-            nrx, src.tostring().c_str(), dest.tostring().c_str(), unsigned(rx.dstif), bind_addr.tostring().c_str());
+    origin_t origin = Direct;
+    if(manager->ifmap.is_lo(rx.dstif) && dest.compare(lo_mcast_addr.addr,false)==0) {
+        // packet forwarded by a local PVA peer (maybe us) as IPv4 local multicast
+        origin = Forward;
```

Comment:
@FreddieAkeroyd This is the change to detect a "forwarded" packet, arriving via a loopback interface, with the (multicast) destination address of `224.0.0.128`.  This is distinct from eg. a packet arriving by loopback destined for `127.0.0.1` or `127.255.255.255`.

With this PR, it is possible to achieve the same effect as the Linux specific `127.255.255.255` in a portable way.

```sh
EPICS_PVA_ADDR_LIST=127.0.0.1 \
EPICS_PVA_AUTO_ADDR_LIST=NO \
  ./bin/linux-x86_64/pvxget from:ioc1 from:ioc2
```

However, this will need some interop testing to make sure that other implementations treat this situation sensibly.